### PR TITLE
Display message when next execution widget doesn't contains any executions

### LINKED ISSUE #83: Update widget with message when no execution are founds
### What feature do you want to see added?

Update widget with message when not execution are founds

To be consistent with other widget (like build queue) that display `No builds in the queue.`

Screenshot from 2024-01-13 13-47-26

### Upstream changes

_No response_

### Are you interested in contributing this feature?

_No response_

### PR DESCRIPTION
**Pull request by l3002**
### Description

Fixes #83 

Changed the Jelly Script for Next Execution Widget on the homepage sidebar for maintaining continuity with the other widget and representing a case where no executions are in process.

Output after change:

Screenshot 2024-01-15 083500

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/resources/hudson/plugins/nextexecutions/NextExecutionsWidget/index.jelly [hunk 1]
```diff
@@ -2,12 +2,19 @@
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
 <j:if test="${it.showWidget()}">
   <l:pane width="2" title="${it.widgetName}" id="${it.widgetId}">
-    <j:forEach var="w" items="${it.builds}">
+    <j:if test="${it.builds() == null}"> 
       <tr>
-        <td class="pane"><a tooltip="${w.name}" href="${w.url}">${w.name}</a></td>
-        <td class="pane" tooltip="${w.timeToGo}">${w.date}</td>
+        <td class="pane">No execution in process.</td>
```
Comment (jonesbusy) [inline src/main/resources/hudson/plugins/nextexecutions/NextExecutionsWidget/index.jelly:7]: This must support i18n and added on `https://github.com/jenkinsci/next-executions-plugin/blob/main/src/main/resources/hudson/plugins/nextexecutions/Messages.properties`

I think also better phrasing would be `No next executions found.`

**Review by jonesbusy — CHANGES_REQUESTED** [review]

**Comment by jonesbusy** [comment]
Thanks for your PR.

Can you also do the same for the other Widget in this plugin ?

- Parametrized next execution (for jobs with parameters)
- Widget on the node page (for jobs stick to a node)
- Possible next execution (for jobs with SCM polling that are triggered on SCM change)

Thanks

File: src/main/resources/hudson/plugins/nextexecutions/NextExecutionsWidget/index.jelly (hunk 1 above)
Comment (l3002) [reply src/main/resources/hudson/plugins/nextexecutions/NextExecutionsWidget/index.jelly:7]: Sorry, I'm a completely new to jelly script. Didn't realize that I would also have to add it to https://github.com/jenkinsci/next-executions-plugin/blob/main/src/main/resources/hudson/plugins/nextexecutions/Messages.properties.

Also, I was trying to find some sources from where I can learn more about jelly script but didn't find much online and the information was in bits and pieces. Do you have any recommendation or good sources from where I can learn more about it?

**Comment by l3002** [comment]
> Thanks for your PR.
> 
> Can you also do the same for the other Widget in this plugin ?
> 
> * Parametrized next execution (for jobs with parameters)
> * Widget on the node page (for jobs stick to a node)
> * Possible next execution (for jobs with SCM polling that are triggered on SCM change)
> 
> Thanks

Sure. I'll try my best.

File: src/main/resources/hudson/plugins/nextexecutions/NextExecutionsWidget/index.jelly (hunk 1 above)
Comment (jonesbusy) [reply src/main/resources/hudson/plugins/nextexecutions/NextExecutionsWidget/index.jelly:7]: https://www.jenkins.io/doc/developer/internationalization/i18n-jelly-views/

Comment (l3002) [reply src/main/resources/hudson/plugins/nextexecutions/NextExecutionsWidget/index.jelly:7]: Thanks a lot.

**Comment by l3002** [comment]
Hi @jonesbusy, Sorry I was a bit busy last week, I'll wrap this up by the end of this week.

**Comment by l3002** [comment]
@jonesbusy, I've update the widget to support i18n for the next execution widget on sidebar. Also, I believe it would be best if we create a new issue for the following:

Parametrized next execution (for jobs with parameters)
Widget on the node page (for jobs stick to a node)
Possible next execution (for jobs with SCM polling that are triggered on SCM change)

as I would require a lot more time for these three. It's been a while since I have worked with Jenkins and in order to test these changes I'll have to read the developers guide for Jenkins too. Also, I am having a few issues with the test instance too.

**Comment by jonesbusy** [comment]
> @jonesbusy, I've updated the next execution widget to support i18n. Also, I believe it would be best if we create a new issue for the following:
> 
> Parametrized next execution (for jobs with parameters) Widget on the node page (for jobs stick to a node) Possible next execution (for jobs with SCM polling that are triggered on SCM change)
> 
> as I would require a lot more time for these three. It's been a while since I have worked with Jenkins and in order to test these changes I'll have to read the developers guide for Jenkins too. Also, I am having a few issues with the test instance too.

It's ok for me, just resolve the last review comment and we should be ok to merge.

Thanks

**Comment by jonesbusy** [comment]
Hi! Any news here ? It's just matter of removing the `.factorypath` and we are good to go.

Thanks

**Comment by l3002** [comment]
Hi @jonesbusy, I'm extremely sorry, I've been really busy with my daytime work. I'll add a commit right away.

**Review by jonesbusy — APPROVED** [review]

**Comment by gandadil** [comment]
Not working for me after this new version released today
image

I had a plenty of jobs showing the next executions. Reverting to previous version resolve to me

**Comment by jonesbusy** [comment]
I have the same. Sorry for the inconveniance, I didn't do interactive test on this PR.

rollback

I will rollback and perform a release.

Thanks for raising it

**Comment by l3002** [comment]
@jonesbusy and @gandadil, I'm really sorry for the inconvenience caused. I'm not sure why is this issue occurring. I'll try to figure it out on my end.

**Comment by gandadil** [comment]
I can test a new version if need, I have 3 Jenkins instances all with this problem, 2 Linux and 1 windows. All with 2.446 Jenkins version

**Comment by jonesbusy** [comment]
Thanks @gandadil I have also many instance that I can test. I would prefer in the future we detect such issue before (The plugin miss of lot of tests, there is only 20% coverage...)

This issue could be detected via HTML unit https://www.jenkins.io/doc/developer/testing/ if jelly view is the problem (I'm not sure yet)

**Pull request merged**